In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import seaborn as sns 
import matplotlib.pyplot as plt

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [2]:
data = pd.read_csv("../input/Billionaires Statistics Dataset.csv", index_col="rank")

In [3]:
data.head(5)

finalWorth               category                personName   age  \
rank                                                                      
1         211000       Fashion & Retail  Bernard Arnault & family  74.0   
2         180000             Automotive                 Elon Musk  51.0   
3         114000             Technology                Jeff Bezos  59.0   
4         107000             Technology             Larry Ellison  78.0   
5         106000  Finance & Investments            Warren Buffett  92.0   

            country    city              source             industries  \
rank                                                                     
1            France   Paris                LVMH       Fashion & Retail   
2     United States  Austin       Tesla, SpaceX             Automotive   
3     United States  Medina              Amazon             Technology   
4     United States   Lanai              Oracle             Technology   
5     United States   Omaha  Berkshire Hathaway  Finance & Investments   

     countryOfCitizenship                      organization  ...  \
rank                                                         ...   
1                  France  LVMH Moët Hennessy Louis Vuitton  ...   
2           United States                             Tesla  ...   
3           United States                            Amazon  ...   
4           United States                            Oracle  ...   
5           United States    Berkshire Hathaway Inc. (Cl A)  ...   

      cpi_change_country           gdp_country  \
rank                                             
1                    1.1   $2,715,518,274,227    
2                    7.5  $21,427,700,000,000    
3                    7.5  $21,427,700,000,000    
4                    7.5  $21,427,700,000,000    
5                    7.5  $21,427,700,000,000    

     gross_tertiary_education_enrollment  \
rank                                       
1                                   65.6   
2                                   88.2   
3                                   88.2   
4                                   88.2   
5                                   88.2   

     gross_primary_education_enrollment_country life_expectancy_country  \
rank                                                                      
1                                         102.5                    82.5   
2                                         101.8                    78.5   
3                                         101.8                    78.5   
4                                         101.8                    78.5   
5                                         101.8                    78.5   

     tax_revenue_country_country total_tax_rate_country population_country  \
rank                                                                         
1                           24.2                   60.7         67059887.0   
2                            9.6                   36.6        328239523.0   
3                            9.6                   36.6        328239523.0   
4                            9.6                   36.6        328239523.0   
5                            9.6                   36.6        328239523.0   

     latitude_country longitude_country  
rank                                     
1           46.227638          2.213749  
2           37.090240        -95.712891  
3           37.090240        -95.712891  
4           37.090240        -95.712891  
5           37.090240        -95.712891  

[5 rows x 34 columns]

In [4]:
data.describe()

finalWorth          age    birthYear   birthMonth     birthDay  \
count    2640.000000  2575.000000  2564.000000  2564.000000  2564.000000   
mean     4623.787879    65.140194  1957.183307     5.740250    12.099844   
std      9834.240939    13.258098    13.282516     3.710085     9.918876   
min      1000.000000    18.000000  1921.000000     1.000000     1.000000   
25%      1500.000000    56.000000  1948.000000     2.000000     1.000000   
50%      2300.000000    65.000000  1957.000000     6.000000    11.000000   
75%      4200.000000    75.000000  1966.000000     9.000000    21.000000   
max    211000.000000   101.000000  2004.000000    12.000000    31.000000   

       cpi_country  cpi_change_country  gross_tertiary_education_enrollment  \
count  2456.000000         2456.000000                          2458.000000   
mean    127.755204            4.364169                            67.225671   
std      26.452951            3.623763                            21.343426   
min      99.550000           -1.900000                             4.000000   
25%     117.240000            1.700000                            50.600000   
50%     117.240000            2.900000                            65.600000   
75%     125.080000            7.500000                            88.200000   
max     288.570000           53.500000                           136.600000   

       gross_primary_education_enrollment_country  life_expectancy_country  \
count                                 2459.000000              2458.000000   
mean                                   102.858520                78.122823   
std                                      4.710977                 3.730099   
min                                     84.700000                54.300000   
25%                                    100.200000                77.000000   
50%                                    101.800000                78.500000   
75%                                    102.600000                80.900000   
max                                    142.100000                84.200000   

       tax_revenue_country_country  total_tax_rate_country  \
count                  2457.000000             2458.000000   
mean                     12.546235               43.963344   
std                       5.368625               12.145296   
min                       0.100000                9.900000   
25%                       9.600000               36.600000   
50%                       9.600000               41.200000   
75%                      12.800000               59.100000   
max                      37.200000              106.300000   

       population_country  latitude_country  longitude_country  
count        2.476000e+03       2476.000000        2476.000000  
mean         5.102053e+08         34.903592          12.583156  
std          5.542447e+08         17.003497          86.762989  
min          3.801900e+04        -40.900557        -106.346771  
25%          6.683440e+07         35.861660         -95.712891  
50%          3.282395e+08         37.090240          10.451526  
75%          1.366418e+09         40.463667         104.195397  
max          1.397715e+09         61.924110         174.885971

In [5]:
country_names =data["country"].value_counts() #List of how many billionaires there are in the country

In [6]:
country_names 

United States           754
China                   523
India                   157
Germany                 102
United Kingdom           82
                       ... 
Portugal                  1
Georgia                   1
Eswatini (Swaziland)      1
Uzbekistan                1
Armenia                   1
Name: country, Length: 78, dtype: int64

> > > > >  ****10 countries with the most billionaires****

In [7]:
data_100 = data.loc[:100,["finalWorth","category","country"]]


In [8]:
data_100_category = data_100["category"].value_counts()

> > > > > ****Most Richest 100 person in the world and their categories****

In [9]:
data_usa = data[data["country"]== "United States"] #We focus on billionaires based in United States


In [10]:
data_usa_category = data_usa["category"].value_counts()
data_usa_category

Finance & Investments         190
Technology                    141
Food & Beverage                74
Fashion & Retail               63
Real Estate                    51
Media & Entertainment          37
Energy                         35
Sports                         35
Healthcare                     34
Manufacturing                  28
Service                        19
Diversified                    13
Automotive                     12
Gambling & Casinos              6
Telecom                         5
Construction & Engineering      5
Logistics                       5
Metals & Mining                 1
Name: category, dtype: int64

> > > > > ****All Billionaires in the USA and their categories****

In [11]:
data_usa_city = data_usa["city"].value_counts()
print(data_usa_city)
data_usa_city.info()

New York             99
San Francisco        37
Los Angeles          34
Palm Beach           21
Dallas               20
                     ..
Steamboat Springs     1
Westlake              1
Coconut Grove         1
Short Hills           1
Pottsville            1
Name: city, Length: 268, dtype: int64
<class 'pandas.core.series.Series'>
Index: 268 entries, New York to Pottsville
Series name: city
Non-Null Count  Dtype
--------------  -----
268 non-null    int64
dtypes: int64(1)
memory usage: 4.2+ KB


268 city exist, I'll try .head(20)

> > > > > **The 20 cities has most billionaires in usa**

In [12]:
data.head(5)

finalWorth               category                personName   age  \
rank                                                                      
1         211000       Fashion & Retail  Bernard Arnault & family  74.0   
2         180000             Automotive                 Elon Musk  51.0   
3         114000             Technology                Jeff Bezos  59.0   
4         107000             Technology             Larry Ellison  78.0   
5         106000  Finance & Investments            Warren Buffett  92.0   

            country    city              source             industries  \
rank                                                                     
1            France   Paris                LVMH       Fashion & Retail   
2     United States  Austin       Tesla, SpaceX             Automotive   
3     United States  Medina              Amazon             Technology   
4     United States   Lanai              Oracle             Technology   
5     United States   Omaha  Berkshire Hathaway  Finance & Investments   

     countryOfCitizenship                      organization  ...  \
rank                                                         ...   
1                  France  LVMH Moët Hennessy Louis Vuitton  ...   
2           United States                             Tesla  ...   
3           United States                            Amazon  ...   
4           United States                            Oracle  ...   
5           United States    Berkshire Hathaway Inc. (Cl A)  ...   

      cpi_change_country           gdp_country  \
rank                                             
1                    1.1   $2,715,518,274,227    
2                    7.5  $21,427,700,000,000    
3                    7.5  $21,427,700,000,000    
4                    7.5  $21,427,700,000,000    
5                    7.5  $21,427,700,000,000    

     gross_tertiary_education_enrollment  \
rank                                       
1                                   65.6   
2                                   88.2   
3                                   88.2   
4                                   88.2   
5                                   88.2   

     gross_primary_education_enrollment_country life_expectancy_country  \
rank                                                                      
1                                         102.5                    82.5   
2                                         101.8                    78.5   
3                                         101.8                    78.5   
4                                         101.8                    78.5   
5                                         101.8                    78.5   

     tax_revenue_country_country total_tax_rate_country population_country  \
rank                                                                         
1                           24.2                   60.7         67059887.0   
2                            9.6                   36.6        328239523.0   
3                            9.6                   36.6        328239523.0   
4                            9.6                   36.6        328239523.0   
5                            9.6                   36.6        328239523.0   

     latitude_country longitude_country  
rank                                     
1           46.227638          2.213749  
2           37.090240        -95.712891  
3           37.090240        -95.712891  
4           37.090240        -95.712891  
5           37.090240        -95.712891  

[5 rows x 34 columns]

In [13]:
category_list = list(data.category.unique())
category_list

['Fashion & Retail',
 'Automotive',
 'Technology',
 'Finance & Investments',
 'Media & Entertainment',
 'Telecom',
 'Diversified',
 'Food & Beverage',
 'Logistics',
 'Gambling & Casinos',
 'Manufacturing',
 'Real Estate',
 'Metals & Mining',
 'Energy',
 'Healthcare',
 'Service',
 'Construction & Engineering',
 'Sports']

In [14]:
data.finalWorth = data.finalWorth.astype(float)
worth_average = []
for i in category_list:
    x = data[data['category']==i]
    worth_ratio = sum(x.finalWorth)/len(x)
    worth_average.append(worth_ratio)
data2 = pd.DataFrame({'category_list': category_list,'worth_average':worth_average})
new_index = (data2['worth_average'].sort_values(ascending=False)).index.values
new_data = data2.reindex(new_index)

In [15]:
data3 =data.dropna(axis="index",how="any",inplace=False) # ı created new df because so many nan values was exist

In [16]:
data3.head() 

finalWorth               category      personName   age        country  \
rank                                                                           
2       180000.0             Automotive       Elon Musk  51.0  United States   
3       114000.0             Technology      Jeff Bezos  59.0  United States   
4       107000.0             Technology   Larry Ellison  78.0  United States   
5       106000.0  Finance & Investments  Warren Buffett  92.0  United States   
6       104000.0             Technology      Bill Gates  67.0  United States   

        city              source             industries countryOfCitizenship  \
rank                                                                           
2     Austin       Tesla, SpaceX             Automotive        United States   
3     Medina              Amazon             Technology        United States   
4      Lanai              Oracle             Technology        United States   
5      Omaha  Berkshire Hathaway  Finance & Investments        United States   
6     Medina           Microsoft             Technology        United States   

                         organization  ...  cpi_change_country  \
rank                                   ...                       
2                               Tesla  ...                 7.5   
3                              Amazon  ...                 7.5   
4                              Oracle  ...                 7.5   
5      Berkshire Hathaway Inc. (Cl A)  ...                 7.5   
6     Bill & Melinda Gates Foundation  ...                 7.5   

               gdp_country gross_tertiary_education_enrollment  \
rank                                                             
2     $21,427,700,000,000                                 88.2   
3     $21,427,700,000,000                                 88.2   
4     $21,427,700,000,000                                 88.2   
5     $21,427,700,000,000                                 88.2   
6     $21,427,700,000,000                                 88.2   

     gross_primary_education_enrollment_country life_expectancy_country  \
rank                                                                      
2                                         101.8                    78.5   
3                                         101.8                    78.5   
4                                         101.8                    78.5   
5                                         101.8                    78.5   
6                                         101.8                    78.5   

     tax_revenue_country_country total_tax_rate_country population_country  \
rank                                                                         
2                            9.6                   36.6        328239523.0   
3                            9.6                   36.6        328239523.0   
4                            9.6                   36.6        328239523.0   
5                            9.6                   36.6        328239523.0   
6                            9.6                   36.6        328239523.0   

     latitude_country longitude_country  
rank                                     
2            37.09024        -95.712891  
3            37.09024        -95.712891  
4            37.09024        -95.712891  
5            37.09024        -95.712891  
6            37.09024        -95.712891  

[5 rows x 34 columns]

In [17]:
data3.info() #in the new df we have 238 rows for each column

<class 'pandas.core.frame.DataFrame'>
Int64Index: 238 entries, 2 to 2540
Data columns (total 34 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   finalWorth                                  238 non-null    float64
 1   category                                    238 non-null    object 
 2   personName                                  238 non-null    object 
 3   age                                         238 non-null    float64
 4   country                                     238 non-null    object 
 5   city                                        238 non-null    object 
 6   source                                      238 non-null    object 
 7   industries                                  238 non-null    object 
 8   countryOfCitizenship                        238 non-null    object 
 9   organization                                238 non-null    object 
 10  selfMade     

In [18]:
data3.country.unique()  # and 238 billionaires in usa, others deleted.

array(['United States'], dtype=object)

**kernel will continue...**